packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1826,6 +1828,11 @@ class StackViewport extends Viewport {
     this._throwIfDestroyed();
 
     this.imageIds = imageIds;
+    this.imageIdsMap.clear();
+    imageIds.forEach((imageId, index) => {
+      this.imageIdsMap.set(imageId, index);
+      this.imageIdsMap.set(imageIdToURI(imageId), index);
```

Comment:
so this seems like it is not just imageIds, but also imageURIs too, so I guess some other variable naming, `imageKeyToIndexMap` ?